src/jesse_json_path.erl

```diff
@@ -9,13 +9,17 @@
 %% API for path management
 -export([new/0, new/1, push/2, pop/1, to_string/1]).
 
+-ifdef(deprecated_types).
+-type dict() :: dict:dict().
+-endif.
+
 -type elem_key_type() :: atom | binary | string | undefined.
 -type elem_type() :: list | elem_key_type().
 -type kvc_obj() :: kvc_obj_node() | [kvc_obj_node()] | list().
 -type kvc_key() :: binary() | atom() | string().
 -type proplist() :: [{kvc_key(), kvc_obj()}].
 -type kvc_obj_node() :: proplist() | {struct, proplist()}
-                      | dict:dict() | gb_tree:tree() | term().
+                      | dict() | gb_tree:tree() | term().
```

Comment:
@zdaniel86 to support my proposal to change to `gb_treeS`:

```
$ dialyzer --src src
  Checking whether the PLT /Users/aturkin/.dialyzer_plt is up-to-date... yes
  Proceeding with analysis...
jesse_schema_validator.erl:398: The pattern 'true' can never match the type 'ok'
jesse_schema_validator.erl:399: The pattern 'false' can never match the type 'ok'
Unknown types:
  gb_tree:tree/0
  jesse:accumulator/0
  json_schema_path:path/0
  json_schema_validator:error/0
 done in 0m1.74s
```

so dialyzer is unhappy with `gb_tree:` and happy with `gb_treeS`.